lib/path-handlers.js

```diff
@@ -47,6 +50,54 @@ function get_path(req, res, next) {
     });
 }
 
+
+function get_dir(req, res, next) {
+    var directories = [], juttles = [], relative_path, abs_path;
+
+
+    Promise.try(function() {
+        // massage the heck out of these paths
+        relative_path = req.query.path || '.';
+        relative_path = relative_path[0] === '/' ? relative_path.substring(1) : relative_path;
+        relative_path = path.normalize(relative_path);
```

Comment:
I'd consider entirely banning the use of /. and  /.. other than possibly a single /.. at the very end of a path if you plan on using it to reach a parent directory.